### PR TITLE
Feature/user route

### PR DESCRIPTION
**Pull request by lazy-val**
Implemented routes for `user`. Read carefully the `PUT` method.

### CODE REVIEW

File: routes/user/router.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+      user.email = req.body.email || user.email;
+      user.freelancer = req.body.freelancer || user.freelancer;
+      if (req.body.freelancer) {
+        utils.checkFreelancerExistsAndIsNotClaimed(req.body.freelancer, function(exists) {
+          if (exists) {
+            user.save(function (err, updatedUser) {
+              if (err) res.status(400).json(utils.formatErrorMessage(err));
+
+              // update the state of the freelancer
+              Freelance.update({ _id: updatedUser.freelancer }, { $set: { state: 'verified' }}, function(err, raw) {
```
Comment (larabr) [inline routes/user/router.js:89]: This is meant to be called once the admin approves a claiming request?

**Review by larabr — COMMENTED** [review]
Maybe add some comments on what a route is for, on a high level (like the PUT for user that also handles part of the claim story from what i understood)

File: routes/user/router.js (hunk 1 above)
Comment (aronfiechter) [reply routes/user/router.js:89]: Yes. To approve a claim request, send a `PUT: /user/:username` that adds the `freelancer` field. The router automatically updates also the freelancer.

Comment (aronfiechter) [reply routes/user/router.js:89]: We had comments on the trello card, I'll add them in the code too.

Comment (aronfiechter) [reply routes/user/router.js:89]: Actually no, I'll add them later, no need to create conflicts now that the router is being modified on `feature/login-route`

Comment (larabr) [reply routes/user/router.js:89]: i dont think comments will be troublesome to merge, but as you wish

**Review by larabr — APPROVED** [review]

**Pull request merged**